source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,200 @@
+        super().update(dt, force_recompute)
+
+    """
+    Implementation.
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor handles and internal buffers.
+
+        This function creates handles and registers the provided data types with the replicator registry to
+        be able to access the data from the sensor. It also initializes the internal buffers to store the data.
+
+        Raises:
+            RuntimeError: If the number of imu prims in the view does not match the number of environments.
+            RuntimeError: If the imu prim is not a RigidBodyPrim
+        """
+        # Initialize parent class
+        super()._initialize_impl()
+        # create simulation view
+        self._physics_sim_view = physx.create_simulation_view(self._backend)
+        self._physics_sim_view.set_subspace_roots("/")
+        # check if the prim at path is a rigid prim
+        prim = sim_utils.find_first_matching_prim(self.cfg.prim_path)
+        if prim is None:
+            raise RuntimeError(f"Failed to find a prim at path expression: {self.cfg.prim_path}")
+        # check if it is a RigidBody Prim
+        if prim.HasAPI(UsdPhysics.RigidBodyAPI):
+            self._view = self._physics_sim_view.create_rigid_body_view(self.cfg.prim_path.replace(".*", "*"))
+        else:
+            raise RuntimeError(f"Failed to find a RigidBodyAPI for the prim paths: {self.cfg.prim_path}")
+
+        # Create internal buffers
+        self._initialize_buffers_impl()
+
+    def _update_buffers_impl(self, env_ids: Sequence[int]):
+        """Fills the buffers of the sensor data."""
+        # check if self._dt is set (this is set in the update function)
+        if not hasattr(self, "_dt"):
+            raise RuntimeError(
+                "The update function must be called before the data buffers are accessed the first time."
+            )
+        # obtain the poses of the sensors
+        pos_w, quat_w = self._view.get_transforms()[env_ids].split([3, 4], dim=-1)
+        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
+        # store the poses
+        # note: we clone here because the obtained tensors are read-only
+        self._data.pos_w[env_ids] = pos_w + math_utils.quat_rotate(quat_w, self._offset_pos)
+        self._data.quat_w[env_ids] = math_utils.quat_mul(quat_w, self._offset_quat)
+
+        # obtain the velocities of the sensors
+        lin_vel_w, ang_vel_w = self._view.get_velocities()[env_ids].split([3, 3], dim=-1)
+        # store the velocities
+        # note: we clone here because the obtained tensors are read-only
+        self._data.ang_vel_b[env_ids] = math_utils.quat_rotate_inverse(self._data.quat_w[env_ids], ang_vel_w)
+        self._data.lin_acc_b[env_ids] = math_utils.quat_rotate_inverse(
+            self._data.quat_w[env_ids],
+            (lin_vel_w - self._last_lin_vel_w[env_ids]) / max(self._dt, self.cfg.update_period),
+        )
+        self._last_lin_vel_w[env_ids] = lin_vel_w.clone()
```

Comment:
The offset is only used for the pose. It isn't accounted for in calculating the linear and angular velocities/accelerations.

Reply:
true, very good catch

Reply:
But that isn't the same right? From the robot dynamics notes at ETH:

<img width="476" alt="image" src="https://github.com/isaac-sim/IsaacLab/assets/12863862/cd421eca-ea61-4888-aad5-47ad33a3b5c1">

<img width="609" alt="image" src="https://github.com/isaac-sim/IsaacLab/assets/12863862/a1d26aa0-2a16-4575-98b0-b44e2bb432eb">

<img width="612" alt="image" src="https://github.com/isaac-sim/IsaacLab/assets/12863862/7bbea7b6-35ab-450b-8f54-07f6d1070f45">

<img width="628" alt="image" src="https://github.com/isaac-sim/IsaacLab/assets/12863862/7b163573-7ab1-46b2-b472-079180262941">

<img width="616" alt="image" src="https://github.com/isaac-sim/IsaacLab/assets/12863862/5fc0818d-4f63-4047-849a-fc117c87407d">

Reply:
preliminary solution is here but not sure if correct, test not working yet